frontend/src/container/MetricsExplorer/Explorer/__tests__/TimeSeries.test.tsx

```diff
@@ -0,0 +1,190 @@
+	__esModule: true,
+	default: jest.fn().mockReturnValue(
+		<div role="img" aria-label="warning">
+			TimeSeriesView
+		</div>,
+	),
+}));
+
+jest.mock('react-query', () => ({
+	...jest.requireActual('react-query'),
+	useQueryClient: jest.fn().mockReturnValue({
+		invalidateQueries: jest.fn(),
+	}),
+	useQueries: jest.fn().mockImplementation((queries: any[]) =>
+		queries.map(() => ({
+			data: undefined,
+			isLoading: false,
+			isError: false,
+			error: undefined,
+		})),
+	),
+}));
+
+jest.mock('react-redux', () => ({
+	...jest.requireActual('react-redux'),
+	useSelector: jest.fn().mockReturnValue({
+		globalTime: {
+			selectedTime: '5min',
+			maxTime: 1713738000000,
+			minTime: 1713734400000,
+		},
+	}),
+}));
+
+const mockMetric: MetricMetadata = {
+	type: MetricType.SUM,
+	description: 'metric1 description',
+	unit: 'metric1 unit',
+	temporality: Temporality.CUMULATIVE,
+	isMonotonic: true,
+};
+
+const queryClient = new QueryClient();
+
+const mockSetWarning = jest.fn();
+const mockSetIsMetricDetailsOpen = jest.fn();
+const mockSetYAxisUnit = jest.fn();
+
+function renderTimeSeries(
+	overrides: Partial<TimeSeriesProps> = {},
+): RenderResult {
+	return render(
+		<QueryClientProvider client={queryClient}>
+			<TimeSeries
+				showOneChartPerQuery={false}
+				setWarning={mockSetWarning}
+				areAllMetricUnitsSame={false}
+				isMetricUnitsLoading={false}
+				metricUnits={[]}
+				metricNames={[]}
+				metrics={[]}
+				isMetricUnitsError={false}
+				handleOpenMetricDetails={mockSetIsMetricDetailsOpen}
+				yAxisUnit="count"
+				setYAxisUnit={mockSetYAxisUnit}
+				// eslint-disable-next-line react/jsx-props-no-spreading
+				{...overrides}
+			/>
+		</QueryClientProvider>,
+	);
+}
+
+describe('TimeSeries', () => {
+	it('should render a warning icon when a metric has no unit among multiple metrics', () => {
+		const { container } = renderTimeSeries({
+			metricUnits: ['', 'count'],
+			metricNames: ['metric1', 'metric2'],
+			metrics: [undefined, undefined],
+		});
+
+		const alertIcon = container.querySelector('.no-unit-warning') as HTMLElement;
+		fireEvent.mouseOver(alertIcon);
+		return waitFor(() =>
+			expect(
+				screen.getByText(/This metric does not have a unit/i),
+			).toBeInTheDocument(),
+		);
+	});
+
+	it('clicking on warning icon tooltip should open metric details modal', async () => {
+		const { container } = renderTimeSeries({
+			metricUnits: ['', 'count'],
+			metricNames: ['metric1', 'metric2'],
+			metrics: [mockMetric, mockMetric],
+			yAxisUnit: 'seconds',
+		});
+
+		const alertIcon = container.querySelector('.no-unit-warning') as HTMLElement;
+		fireEvent.mouseOver(alertIcon);
+
+		const metricDetailsLink = await screen.findByText(/metric details/i);
+		fireEvent.click(metricDetailsLink);
+
+		expect(mockSetIsMetricDetailsOpen).toHaveBeenCalledWith('metric1');
+	});
+
+	it.skip('shows Save unit button when metric had no unit but one is selected', () => {
```

Comment:
why skip?

Reply:
We are skipping the tests for the save functionality as we have disabled the save feature itself. As we want to migrate the metrics explorer APIs from v1 to v2 before enabling it. 
I've a separate ticket to enable there after the migration: https://github.com/SigNoz/signoz/issues/9861